src/evict/evict_inline.h

```diff
@@ -628,6 +628,140 @@ __evict_check_user_ok_with_eviction(WT_SESSION_IMPL *session, bool interruptible
     return (true);
 }
 
+/* !!!
+ * __evict_is_session_cache_trigger_tolerant --
+ *     Check if the session is cache tolerant for the configured trigger values. If tolerant,
+ *     session will not perform eviction.
+ *
+ * Cache tolerance is cache's ability to let application threads perform workload operations above
+ *     trigger levels. Cache tolerance is applicable only for dirty_trigger and updates_trigger.
+ *     Cache tolerance is expressed in % with respect to the trigger level. If cache tolerance is
+ *     configured as 20%, and dirty trigger is 20%, then application threads will be incrementally
+ *     involve in eviction as dirty content increases from 20% - 24%.
+ *
+ * All the application threads will perform eviction only when the dirty content reaches 24% of
+ *     cache.
+ *
+ * Based on the % of cache usage above trigger level with respect to tolerance level,
+ *     incrementally involve the app threads for eviction.
+ *
+ *     Usage between 00% - 20% --> No application threads involve in eviction.
+ *     Usage between 20% - 40% --> 20% of application threads involve in eviction.
+ *     Usage between 40% - 60% --> 40% of application threads involve in eviction.
+ *     Usage between 60% - 80% --> 60% of application threads involve in eviction.
+ *     Usage between 80% - 100% --> 80% of application threads involve in eviction.
+ *     Usage above 100%, above tolerance level --> involve all application threads.
+ */
+static WT_INLINE bool
+__evict_is_session_cache_trigger_tolerant(WT_SESSION_IMPL *session, uint8_t cache_tolerance)
+{
+    double dirty_trigger = __wt_atomic_load_double(&S2C(session)->evict->eviction_dirty_trigger);
+    uint64_t bytes_dirty_trigger = 0, bytes_max = 0;
+    uint64_t bytes_dirty = 0, bytes_dirty_tolerance = 0, bytes_over_dirty_trigger = 0;
+
+    bytes_max = S2C(session)->cache_size + 1;
+    bytes_dirty = __wt_cache_dirty_leaf_inuse(S2C(session)->cache);
+    bytes_dirty_trigger = (uint64_t)(dirty_trigger * bytes_max) / 100;
```

Comment:
The value inside `eviction_dirty_trigger` is already a calculated value.

`   WT_RET(__wt_config_gets(session, cfg, "eviction_dirty_trigger", &cval));
    evict->eviction_dirty_trigger = (double)cval.val;
    WT_RET(__evict_config_abs_to_pct(session, &(evict->eviction_dirty_trigger),
      "eviction dirty trigger", conn->cache_size, shared));`